src/meta.rs

```diff
@@ -0,0 +1,176 @@
+                            .and_then(|object_output| {
+                                // Deserialize the object output and wrap in an `MetaConfigObject`
+                                object_output
+                                    .body
+                                    .unwrap()
+                                    .concat2()
+                                    .map_err(|e| {
+                                        error!("concatenating body: {:?}", e);
+                                        ()
+                                    })
+                                    .map(move |bytes| {
+                                        let result = String::from_utf8(bytes).unwrap();
+                                        let parts: Vec<&str> = file_key_clone
+                                            .trim_start_matches("minsql/meta/")
+                                            .split("/")
+                                            .collect();
+                                        let meta_obj = match (parts.len(), parts[0]) {
+                                            (2, "logs") => match serde_json::from_str(&result) {
+                                                Ok(t) => MetaConfigObject::Log(t),
+                                                Err(_) => MetaConfigObject::Unknown,
+                                            },
+                                            (2, "datastores") => {
+                                                match serde_json::from_str(&result) {
+                                                    Ok(t) => MetaConfigObject::DataStore(t),
+                                                    Err(_) => MetaConfigObject::Unknown,
+                                                }
+                                            }
+                                            _ => MetaConfigObject::Unknown,
+                                        };
+                                        meta_obj
+                                    })
+                            })
+                    })
+                    .collect()
+            })
+            .map_err(|e| {
+                println!("mapping contents to structs:  {:?}", e);
+                ()
+            })
+            .map(move |result_meta_objects: Vec<MetaConfigObject>| {
+                //get a write lock on config
+                let mut cfg = main_cfg.write().unwrap();
+                //time to update the configuration!
+                for mco in result_meta_objects {
+                    match mco {
+                        MetaConfigObject::Log(l) => {
+                            cfg.log.insert(l.clone().name.unwrap(), l);
+                        }
+                        MetaConfigObject::DataStore(ds) => {
+                            cfg.datastore.insert(ds.clone().name.unwrap(), ds);
+                        }
+                        _ => (),
+                    }
+                }
+            })
```

Comment:
This map also is better separated out into a function.

Reply:
It will be refactored but there's still improvements that need to be made on top of this pipeline

Reply:
Ok.